src/java/htsjdk/samtools/util/SequenceUtil.java

```diff
@@ -307,6 +357,20 @@ public static String makeSoftClipCigar(final int clipLength) {
         return Integer.toString(clipLength) + "S";
     }
 
+    /**
+     * Helper method to handle the various use cases of base comparison.
+     */
```

Comment:
This could probably use a little doc on the parameters.  Specifically that `matchAmbigious` causes the match to return true when the read base is a subset of the possible reference bases, but not the other way around.  Same comment about name implying reflexiveness.